python/tvm/contrib/popen_pool.py

```diff
@@ -153,6 +160,11 @@ def _start(self):
         self._reader = os.fdopen(main_read, "rb")
         self._writer = os.fdopen(main_write, "wb")
 
+        # init
+        if self._initializer is not None:
+            self.send(self._initializer, self._initargs)
+            self.recv()
```

Comment:
I am not sure about the logic here. Looks like the workflow is:

* `PopenWorker.send` is the entry point, which calls `PopenWorker._start` if the worker is not available
* with this logic, the `PopenWorker._start` method will call back to `PopenWorker.send` to send the initializer to the worker process.

I didn't see any problem with the workflow itself, but the cyclic call is a bit indirect. Would you mind if we move these 4 lines into the `if self._proc is None` check?

Reply:
done